src/Raven.Server/Documents/PeriodicBackup/Retention/S3RetentionPolicyRunner.cs

```diff
@@ -38,11 +38,16 @@ protected override string GetFolderName(string folderPath)
             return folderPath.Substring(0, folderPath.Length - 1);
         }
 
-        protected override async Task<string> GetFirstFileInFolder(string folder)
+        protected override async Task<GetBackupFolderFilesResult> GetBackupFilesInFolder(string folder)
         {
+            var backupFiles = new GetBackupFolderFilesResult();
             // backups are ordered in lexicographical order
             var files = await _client.ListObjects(folder, null, false, 1);
-            return files.FileInfoDetails?.Select(x => x.FullPath).FirstOrDefault();
+
+            backupFiles.FirstFile = files.FileInfoDetails?.Select(x => x.FullPath).FirstOrDefault();
+            backupFiles.LastFile = files.FileInfoDetails?.Select(x => x.FullPath).LastOrDefault();
```

Comment:
This wouldn't work, you are taking only one result from `var files = await _client.ListObjects(folder, null, false, 1);`
Need to check in the S3 docs, but maybe we can set the as prefix the last retention date and if we get a result we _can't_ delete this folder.

Reply:
done